### PR TITLE
NH-31695: update liboboe 12.0.0 and include amazon linux distro for r…

### PR DESCRIPTION
**Pull request by xuan-cao-swi**
…egression test

## Why?
liboboe 12.0.0 de-supported centos 7. Using amazonlinux as the replacement

## Impact?
1. included new docker-compose for arm build testing (only for local testing)
2. changed the liboboe version to 12.0.0
3. updated the naming convention for the binary file
4. included amzlinux dockerfile

### CODE REVIEW

File: lib/rails/generators/solarwinds_apm/install_generator.rb [hunk 1]
```diff
@@ -29,10 +29,10 @@ def print_header
       say "-------------------"
       say ""
       say "SolarWindsAPM Installation Overview:"
-      say "https://documentation.solarwinds.com/en/success_center/swaas/"
+      say "https://documentation.solarwinds.com/en/success_center/observability"
       say ""
       say "More information on instrumenting Ruby applications can be found here:"
-      say "https://documentation.solarwinds.com/en/success_center/swaas/default.htm#cshid=app-add-ruby-agent"
+      say "https://documentation.solarwinds.com/en/success_center/observability/default.htm#cshid=config-ruby-agent"
```
Comment (cheempz) [inline lib/rails/generators/solarwinds_apm/install_generator.rb:35]: Nice catch ;)

File: lib/rails/generators/solarwinds_apm/templates/solarwinds_apm_initializer.rb [hunk 2]
```diff
@@ -46,9 +46,11 @@
...
   #
   # It is recommended to configure the proxy in this file or as SW_APM_PROXY
-  # environment variable. However, the agent's underlying network library will
-  # use a system-wide proxy defined in the environment variables grpc_proxy,
-  # https_proxy or http_proxy if no SolarWindsAPM-specific configuration is set.
+  # environment variable. However, the underlying network library for the Ruby 
+  # Instrumentation Library will use a system-wide proxy defined in the environment 
+  # variables grpc_proxy, https_proxy or http_proxy if no SolarWindsAPM-specific 
+  # configuration is set.
+  #
```
Comment (cheempz) [inline lib/rails/generators/solarwinds_apm/templates/solarwinds_apm_initializer.rb:53]: Another nice catch, I'll make a docs PR to make the corresponding changes!

**Review by cheempz — DISMISSED** [review]
LGTM, thanks @xuan-cao-swi!  Minor curiosity, you've kept the CentOS 7 Dockerfile and test suite in the docker-compose file, which as far as I know liboboe 12.0.0 (and thus Ruby instrumentation) drop support for -- so I'd expect failing tests... are you seeing something different?

File: lib/rails/generators/solarwinds_apm/templates/solarwinds_apm_initializer.rb (hunk 2 above)
Comment (xuan-cao-swi) [reply lib/rails/generators/solarwinds_apm/templates/solarwinds_apm_initializer.rb:53]: Thanks, Erica pm me about the required change.

**Review by cheempz — APPROVED** [review]
👍

**Pull request merged**